src/syslogd.c

```diff
@@ -3271,6 +3301,58 @@ static void logit(char *fmt, ...)
 	fflush(stdout);
 }
 
+static void notifier_add(struct notifiers *newn, const char *program)
+{
+	while (*program && isspace(*program))
+		program++;
+
+	/* Check whether it is accessible, regardless of TOCTOU */
+	if (!access(program, X_OK)) {
+		struct notifier *np;
+		size_t len;
+
+		len = strlen(program);
+
+		np = calloc(1, sizeof(*np) + len +1);
+		if (np) {
+			/* xxx Actually wastes space -- vararray? */
+			np->n_program = (char*)&np[1];
+			memcpy(np->n_program, program, len);
+			SIMPLEQ_INSERT_TAIL(newn, np, n_link);
+		} else
```

Comment:
Please change this part to something as follows, handling errors first, and not trying to be clever:
```c
np = calloc(1, sizeof(*np));
if (!np) {
   ERR(...);
   return;
}
np->n_program = strdup(program);
if (!np->n_program) {
   free(np);
   ERR(...);
   return;
}
```
It's more readable and follows the general error handling style of syslogd.

Reply:
Did that.